src/Subscription/Activity.php

```diff
@@ -45,4 +46,15 @@ public function getPeriod(): \DateInterval
     {
         return $this->period;
     }
+
+    /**
+     * @param Core\Activity $activity
+     * @throws SubscriptionIsForbidden
+     */
+    public function isValid(Core\Activity $activity): void
+    {
+        if ($this->activity !== $activity) {
```

Comment:
You should not expect the same instance, let's use the activity name as an identifier.